gcloud/vision/client.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Client for interacting with the Google Cloud Vision API."""
+
+import json
+from json import JSONEncoder
+
+from gcloud.client import JSONClient
+from gcloud.vision.connection import Connection
+from gcloud.vision.feature import Feature
+from gcloud.vision.image import Image
+
+
+class VisionJSONEncoder(JSONEncoder):
+    def default(self, o):
+        if 'as_dict' in dir(o):
+            return o.as_dict()
+        else:
+            return o.__dict__
```

Comment:
I don't see a problem with this, performance wise.  It might be YAGNI if you don't already have plans to serialize more than one kind of thing using `as_dict`.